infrastructures/kubernetes/src/main/java/org/eclipse/che/workspace/infrastructure/kubernetes/K8sInfraNamespaceWsAttributeValidator.java

```diff
@@ -76,7 +77,14 @@ public void validate(Map<String, String> attributes) throws ValidationException
                 + "')");
       }
 
-      namespaceFactoryProvider.get().checkIfNamespaceIsAllowed(namespace);
+      // the current user is going to be anonymous only during internal scheduled jobs. In those
+      // circumstances, we don't actually need to check the namespace validity, because those
+      // workflows are defined by our code and are therefore "safe". User-initiated workspace
+      // creation or update is never going to succeed using an anonymous user anyway, so again,
+      // skipping this check for anonymous user is safe.
+      if (!EnvironmentContext.getCurrent().getSubject().isAnonymous()) {
+        namespaceFactoryProvider.get().checkIfNamespaceIsAllowed(namespace);
```

Comment:
maybe this https://github.com/eclipse/che/blob/master/infrastructures/kubernetes/src/main/java/org/eclipse/che/workspace/infrastructure/kubernetes/namespace/KubernetesNamespaceFactory.java#L145-L152 could be a better place fo this `if`. The reason - because  `EnvironmentContext.getCurrent().getSubject()` is actually used in that part of the code.
WDYT?

Reply:
I would propose one more alternative way https://github.com/eclipse/che/compare/master...sleshchenko:infraNamespaceValidationFix?expand=1
<- It's not really tested but I assume it should work.
Why I think it makes sense:
- Anonymous must not create workspaces
- Anonymous must not update infra namespace value.
It can update some info in workspace attributes, like stoppedTime but in such case infraNamespace is the same as workspace now have.
My changes should not validate namespace format in such case - because it does not really make sense. If previously we considered infraNamespace as valid and run namespace there - then we should not validate it later, I suppose..

@metlos WDYT?

Reply:
Thanks for the suggestion. Took me some time to realize why this should work but it is a much nicer solution than my hack.